samples/openthread/cli/README.rst

```diff
@@ -51,6 +51,7 @@ User interface
 All the interactions with the application are handled using serial communication.
 See `OpenThread CLI Reference`_ for the list of available serial commands.
 
+Sample by default uses hardware flow control mechanism in serial communication.
```

Comment:
Move this somewhere around where putty is mentioned. (Probably multiple places in this sample)